src/librustc_mir/transform/generator.rs

```diff
@@ -126,6 +127,14 @@ fn self_arg() -> Local {
     Local::new(1)
 }
 
+struct SuspensionPoint {
+    state: u32,
+    resume: BasicBlock,
+    drop: Option<BasicBlock>,
+    storage_liveness: liveness::LocalSet,
```

Comment:
I find this a bit ugly. Can you have a

```Rust
fn make_storage_live(..., point: &SuspensionPoint, terminates_to: BasicBlock) {
}
```

And call it from both collects to avoid the ugly mutation? I think you can.